File: src/main/java/com/natia/secretmod/utils/emeliaAPI/AuctionHouse.java
package com.natia.secretmod.utils.emeliaAPI;

import com.google.gson.JsonArray;
import com.google.gson.JsonObject;

import java.util.concurrent.atomic.AtomicInteger;
import java.util.concurrent.atomic.AtomicReference;

import static com.natia.secretmod.utils.WebUtils.fetch;

public class AuctionHouse {

    public static String authorizable = "gMoThEA8RFDMj@c?KqT@asJ";

    /**
     * Gets Auction House Info
     */
    public static JsonArray getAuction(String query) {
        AtomicReference<JsonArray> array = new AtomicReference<>(new JsonArray());
        fetch("http://localhost:3000/ah?token=" + authorizable + "&" + query, res -> {
            if (res.asJson().get("success").getAsBoolean()) {
                array.set(res.asJson().get("auctions").getAsJsonArray());
            }
        });
        return array.get();
    }

    public static JsonObject getLowest(JsonArray auctions) {
        AtomicInteger lowest = new AtomicInteger();
        AtomicReference<Double> minLowest = new AtomicReference<>(10000000000000d);
        AtomicInteger currAuction = new AtomicInteger(-1);
        auctions.forEach(jsonElement -> {
            currAuction.getAndIncrement();
            if (jsonElement.getAsJsonObject().get("starting_bid").getAsDouble() < minLowest.get()) {
                lowest.set(currAuction.get());
                minLowest.set(jsonElement.getAsJsonObject().get("starting_bid").getAsDouble());
            }
        });

        return auctions.get(lowest.get()).getAsJsonObject();
    }

    public static JsonObject getSecondLowest(JsonArray auctions) {
        AtomicInteger lowest = new AtomicInteger(9);
        AtomicInteger lastLowest = new AtomicInteger(9);

        AtomicReference<Double> minLowest = new AtomicReference<>(10000000000000d);
        AtomicInteger currAuction = new AtomicInteger(-1);
        auctions.forEach(jsonElement -> {
            currAuction.getAndIncrement();
            if (jsonElement.getAsJsonObject().get("starting_bid").getAsDouble() < minLowest.get()) {
                lastLowest.set(lowest.get());
                lowest.set(currAuction.get());
                minLowest.set(jsonElement.getAsJsonObject().get("starting_bid").getAsDouble());
            }
        });

        return auctions.get(lastLowest.get()).getAsJsonObject();
    }

}

File: src/main/kotlin/natia/skytweaks/utils/emeliaAPI/AuctionHouse.kt
package natia.skytweaks.utils.emeliaAPI

import com.google.gson.JsonArray
import com.google.gson.JsonObject
import natia.skytweaks.utils.WebUtils

import java.util.concurrent.atomic.AtomicInteger
import java.util.concurrent.atomic.AtomicReference

object AuctionHouse {

    var authorizable = "gMoThEA8RFDMj@c?KqT@asJ"

    /**
     * Gets Auction House Info
     */
    fun getAuction(query: String): JsonArray {
        val array = AtomicReference(JsonArray())
        WebUtils.fetch("http://localhost:3000/ah?token=$authorizable&$query", { res ->
            if (res.asJson().get("success").getAsBoolean()) {
                array.set(res.asJson().get("auctions").getAsJsonArray())
            }
        })
        return array.get()
    }

    fun getLowest(auctions: JsonArray): JsonObject {
        val lowest = AtomicInteger()
        val minLowest = AtomicReference(10000000000000.0)
        val currAuction = AtomicInteger(-1)
        auctions.forEach { jsonElement ->
            currAuction.getAndIncrement()
            if (jsonElement.asJsonObject.get("starting_bid").asDouble < minLowest.get()) {
                lowest.set(currAuction.get())
                minLowest.set(jsonElement.asJsonObject.get("starting_bid").asDouble)
            }
        }

        return auctions.get(lowest.get()).asJsonObject
    }

    fun getSecondLowest(auctions: JsonArray): JsonObject {
        val lowest = AtomicInteger(9)
        val lastLowest = AtomicInteger(9)

        val minLowest = AtomicReference(10000000000000.0)
        val currAuction = AtomicInteger(-1)
        auctions.forEach { jsonElement ->
            currAuction.getAndIncrement()
            if (jsonElement.asJsonObject.get("starting_bid").asDouble < minLowest.get()) {
                lastLowest.set(lowest.get())
                lowest.set(currAuction.get())
                minLowest.set(jsonElement.asJsonObject.get("starting_bid").asDouble)
            }
        }

        return auctions.get(lastLowest.get()).asJsonObject
    }

}
